test/unit/spec/userMenuSpec.js

```diff
@@ -65,29 +67,86 @@ define(['jquery', 'userMenu', 'narrativeConfig', 'narrativeMocks'], (
-                    document.querySelector('.modal [data-element="signout-warning-body"]')
-                ).toBeNull();
-                jasmine.clock().uninstall();
+        // logout popups
+        const logoutTests = [
+            {
+                desc: 'simulate cancelling signout',
+                button: 'a[data-element="cancel-signout"]',
+                // there should be no 'logout.kbase' event
+                finalTest: (callArgs) => {
+                    return callArgs.every((args) => {
+                        return typeof args[0] !== 'string' || args[0] !== 'logout.kbase';
+                    });
+                },
+            },
+            {
+                desc: 'simulate signout',
+                button: 'a[data-element="signout"]',
+                // the 'logout.kbase' event should have been triggered
+                finalTest: (callArgs) => {
+                    return callArgs.some((args) => {
+                        return (
+                            typeof args[0] === 'string' &&
+                            args[0] === 'logout.kbase' &&
+                            args[1] === true
+                        );
+                    });
+                },
+            },
+        ];
+
+        logoutTests.forEach((test) => {
+            it(`should ${test.desc} when clicking logout and then the modal button`, () => {
+                const $elem = $('<div>');
+                const userMenu = makeTestUserMenu($elem);
+                const modalQuerySelector = '.modal [data-element="signout-warning-body"]';
+
+                spyOn(BootstrapDialog.prototype, 'show').and.callThrough();
+                spyOn(BootstrapDialog.prototype, 'hide').and.callThrough();
+                spyOn(BootstrapDialog.prototype, 'destroy').and.callThrough();
+                spyOn($.fn, 'trigger').and.callThrough();
+
+                return userMenu
+                    .start()
+                    .then(() => {
+                        expect(document.querySelector(modalQuerySelector)).toBeNull();
+
+                        // click the signout button
+                        $elem.find('#signout-button').click();
+                        expect(BootstrapDialog.prototype.show).toHaveBeenCalled();
+                        expect(BootstrapDialog.prototype.hide).not.toHaveBeenCalled();
+                        expect(BootstrapDialog.prototype.destroy).not.toHaveBeenCalled();
+                        return TestUtil.waitForElement(document.body, modalQuerySelector);
+                    })
+                    .then(() => {
+                        // expect to see the modal appear now
+                        expect(document.querySelector(modalQuerySelector)).not.toBeNull();
+
+                        // click the appropriate button for this test
+                        document.querySelector(test.button).click();
+                        expect(BootstrapDialog.prototype.hide).toHaveBeenCalled();
```

Comment:
clicking the button hides the modal, and once it has been hidden, it gets destroyed.